sql/core/src/test/scala/org/apache/spark/sql/test/TestSQLContext.scala

```diff
@@ -34,9 +40,32 @@ private[sql] class TestSparkSession(sc: SparkContext) extends SparkSession(sc) {
     this(new SparkConf)
   }
 
-  @transient
-  override lazy val sessionState: SessionState = new SessionState(self) {
-    override lazy val conf: SQLConf = {
+  class TestSessionState(
```

Comment:
Why do you need to define a new class? seems like u are taking a  bunch of stuff and passing them as is.